src/tree_builder/mod.rs

```diff
@@ -19,9 +19,12 @@ use self::types::*;
 use self::actions::TreeBuilderActions;
 use self::rules::TreeBuilderStep;
 
+use string_cache::QualName;
+
 use tokenizer;
 use tokenizer::{Doctype, Tag};
 use tokenizer::TokenSink;
+use tokenizer::states::{State, RawData, Rcdata, Rawtext, ScriptData, Data, Plaintext};
```

Comment:
Can this be `use tokenizer::states as tok_state;` or something?  Having unqualified tokenizer states in the tree builder might be confusing, although I may have already done it somewhere...